clang/test/SemaObjCXX/sel-address.mm

```diff
@@ -10,6 +10,7 @@ void h() {
   SEL* ps = &s;
 
   @selector(dealloc) = s;  // expected-error {{expression is not assignable}}
+  // expected-note@-1 {{add '*' to dereference it}}
```

Comment:
From my knowledge, this would assign the value of s to the SEL object pointed to by @selector(dealloc). However, in Objective-C, selectors are not pointers, they are special compile-time constructs representing method names, and they are immutable, so you cannot assign values to them. 

Therefore, this syntax is not valid for selectors in Objective-C.

Reply:
I made a check for `isObjCObjectPointerType`, maybe there's a separate one for `@selector` as well?

Reply:
One way I can think of is to do a AST visit and check for `ObjCSelectorExpr` in the child nodes of the `ValueDecl`.

Reply:
AST visitors are rather heavy atm so I’d rather not do that... I think this is fine provided that `*@selector(dealloc) = s` is valid in Objective-C (as I said, I don’t know Objective-C...)

Reply:
Actually, it isn't valid in case of `@selector`, that's why I was thinking of potential solutions. 

Like one of the tests got ignored using `!E->getType()->isObjCObjectPointerType()`, similarly clang would have for selectors as well.

Reply:
Yeah, again, I unfortunately won’t be of much help with this issue in particular due to lack of Objective-C knowledge; if we can’t find a way around this then it *might* be fine to leave this as a false positive (partially depending on how common this case is), but I’d prefer it for someone who actually knows Objective-C to comment on whether that is the case.

The only thing I can think of is we can maybe work off of whatever the type of `@selector(dealloc)` is, provided that that’s anything special?

Reply:
I see a way to handle this:

`E->getType()->isSpecificBuiltinType(clang::BuiltinType::ObjCSel)` 

Unfortunately doesn't work, since from what I understand, we could be checking for a compound expression here for `E`.

Reply:
> we could be checking for a compound expression here for `E`.

Er, I’m candidly not entirely sure what you’re referring to.